src/hdskins/java/com/voxelmodpack/hdskins/resources/texture/ThreadDownloadImageEncrypt.java

```diff
@@ -167,50 +167,32 @@ private State checkLocalCache() {
             if (code != HttpStatus.SC_OK) {
                 return State.NOPE;
             }
-            return checkETag(response) ? State.OK : State.OUTDATED;
+            return State.OK;
         } catch (IOException e) {
             LOGGER.error("Couldn't load skin {} ", imageUrl, e);
             return State.NOPE;
         }
     }
 
-    private boolean checkETag(HttpResponse response) {
-        try {
-            if (Files.isRegularFile(cacheFile)) {
-                String localETag = Files.lines(eTagFile).limit(1).findFirst().orElse("");
-                Header remoteETag = response.getFirstHeader(HttpHeaders.ETAG);
-                // true if no remote etag or does match
-                return remoteETag == null || localETag.equals(remoteETag.getValue());
-            }
-            return false;
-        } catch (IOException e) {
-            // it failed, so re-fetch.
-            return false;
-        }
-    }
-
     private void loadTextureFromServer() {
         LOGGER.debug("Downloading http texture from {} to {}", imageUrl, cacheFile);
         try (MoreHttpResponses resp = MoreHttpResponses.execute(client, new HttpGet(imageUrl))) {
             if (resp.ok()) {
                 // write the image to disk
                 Files.createDirectories(cacheFile.getParent());
-                Files.copy(resp.getInputStream(), cacheFile);
-
-                try (InputStream in = Files.newInputStream(cacheFile)) {
-                    BufferedImage bufferedimage = ImageIO.read(in);
-
-                    // maybe write the etag to disk
-                    Header eTag = resp.getResponse().getFirstHeader(HttpHeaders.ETAG);
-                    if (eTag != null) {
-                        Files.write(eTagFile, Collections.singleton(eTag.getValue()));
-                    }
-
-                    if (imageBuffer != null) {
-                        bufferedimage = imageBuffer.parseUserSkin(bufferedimage);
-                    }
-                    setBufferedImage(bufferedimage);
+                // Files.copy(resp.getInputStream(), cacheFile);
+                byte[] imageBytes = IOUtils.toByteArray(resp.getInputStream());
```

Comment:
Close your streams.

Reply:
Aren't they closed automagically?

Reply:
Nope!

You have to close them manually, or use an auto-closer:

```
try (InputSream stream = resp.getinputStream()) {
   ...
}
// stream is automagically closed
```

Otherwise pull an use `resp.bytes()`

Reply:
Will do. Also am going to create a new file instead of patching a deprecated class...

Reply:
Do I also need to close (or, for that matter, add the try-hard blocks for) things like `InputStream in = new ByteArrayInputStream(imageBytes);`?

Reply:
The ByteArray streams don't actually close, so you don't have to worry about them.
```java
public void close() throws IOException {
}
```